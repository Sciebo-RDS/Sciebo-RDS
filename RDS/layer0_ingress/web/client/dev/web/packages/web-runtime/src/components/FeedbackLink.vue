<template>
  <div>
    <oc-button
      v-oc-tooltip="description"
      type="a"
      :href="href"
      target="_blank"
      appearance="raw"
      :aria-label="ariaLabel"
      aria-describedby="oc-feedback-link-description"
    >
      <oc-icon name="feedback" />
    </oc-button>
    <p id="oc-feedback-link-description" class="oc-invisible-sr" v-text="description" />
  </div>
</template>

<script>
export default {
  name: 'FeedbackLink',
  computed: {
    href() {
      return 'https://owncloud.com/web-design-feedback'
    },
    ariaLabel() {
      return this.$gettext('ownCloud feedback survey')
    },
    description() {
      return this.$gettext(
        "Provide your feedback: We'd like to improve the web design and would be happy to hear your feedback. Thank you! Your ownCloud team."
      )
    }
  }
}
</script>
