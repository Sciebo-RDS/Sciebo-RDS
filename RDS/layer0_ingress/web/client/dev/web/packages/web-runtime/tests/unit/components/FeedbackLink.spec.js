import FeedbackLink from '../../../src/components/FeedbackLink'
import { createLocalVue, mount } from '@vue/test-utils'
import { axe, toHaveNoViolations } from 'jest-axe'
import DesignSystem from 'owncloud-design-system'
const localVue = createLocalVue()
localVue.use(DesignSystem)
const OcTooltip = jest.fn()
expect.extend(toHaveNoViolations)

describe('FeedbackLink component', () => {
  it('has no accessibility violations', async () => {
    const wrapper = mount(FeedbackLink, {
      localVue,
      directives: {
        OcTooltip
      }
    })

    expect(
      await axe(wrapper.html(), {
        rules: {
          region: { enabled: false }
        }
      })
    ).toHaveNoViolations()
    expect(wrapper).toMatchSnapshot()
  })
})
