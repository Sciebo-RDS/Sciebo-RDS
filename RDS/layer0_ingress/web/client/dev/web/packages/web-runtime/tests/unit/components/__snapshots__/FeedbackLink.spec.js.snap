// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackLink component has no accessibility violations 1`] = `
<div><a aria-label="ownCloud feedback survey" href="https://owncloud.com/web-design-feedback" class="oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw" target="_blank" aria-describedby="oc-feedback-link-description"><span class="oc-icon oc-icon-m oc-icon-passive"><svg height="24" width="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false"> <path xmlns="http://www.w3.org/2000/svg" d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 12h-2v-2h2v2zm0-4h-2V6h2v4z"></path> </svg></span></a>
  <p id="oc-feedback-link-description" class="oc-invisible-sr">Provide your feedback: We'd like to improve the web design and would be happy to hear your feedback. Thank you! Your ownCloud team.</p>
</div>
`;
